stock_warehouse_flow_release/models/stock_move.py

```diff
@@ -8,19 +8,17 @@
 class StockMove(models.Model):
     _inherit = "stock.move"
 
-    def _apply_flow_on_action_confirm(self):
-        # Override to not apply the flow configuration on moves to release.
-        # The flow will be applied on the release, not before.
-        if self.rule_id.route_id.available_to_promise_defer_pull:
+    def _apply_flow_on_release(self):
+        if self.rule_id.route_id.apply_flow_on != "on_release":
             return False
-        return super()._apply_flow_on_action_confirm()
+        return self.picking_type_id.code == "outgoing"
 
     def _before_release(self):
         # Apply the flow when releasing the move
         super()._before_release()
         FLOW = self.env["stock.warehouse.flow"]
         move_ids_to_release = []
-        for move in self:
+        for move in self.filtered(lambda m: m._apply_flow_on_release()):
             _move_ids_to_release = FLOW._search_and_apply_for_move(move).ids
```

Comment:
A second iteration is unnecessary if you move the check `_apply_flow_on_release` into the for loop. 
```suggestion
        for move in self:
            if not move._apply_flow_on_release():
                continue
            _move_ids_to_release = FLOW._search_and_apply_for_move(move).ids
```

Reply:
it's done already, please target a line that changed when suggesting comment, so github can flag them as outdated :pray: 
Thanks!

Reply:
Where is it done?

Reply:
@mmequignon it's not showing as done in the diff...

Reply:
My bad, I messed up while checking locally :facepalm: 
It's pushed now